tests/foreman/cli/test_repositories.py

```diff
@@ -47,3 +50,43 @@ def test_positive_custom_products_disabled_by_default(
     assert rhel_contenthost.subscribed
     product_details = rhel_contenthost.run('subscription-manager repos --list')
     assert 'Enabled:   0' in product_details.stdout
+
+
+def test_negative_invalid_repo_fails_publish(
+    module_repository,
+    module_org,
+    target_sat,
+):
+    """Verify that an invalid repository fails when trying to publish in a content view
+
+    :id: 64e03f28-8213-467a-a229-44c8cbfaaef1
+
+    :steps:
+        1. Create custom product and upload repository
+        2. Run Katello commands to make repository invalid
+        3. Create content view and add repository 
+        4. Verify Publish fails 
+
+    :expectedresults: Publishing a content view with an invalid repository fails
+
+    :customerscenario: true
+
+    :BZ: 2032040
+    """
+    repo = module_repository
+    with target_sat.session.shell() as sh:
+            sh.send('foreman-rake console')
+            time.sleep(30)  # sleep to allow time for console to open
+            sh.send(f'root = ::Katello::RootRepository.last')
+            time.sleep(3)  # give enough time for the command to complete
+            sh.send(f'::Katello::Resources::Candlepin::Product.remove_content(root.product.organization.label, root.product.cp_id, root.content_id)')
+            time.sleep(3)
+            sh.send(f'::Katello::Resources::Candlepin::Content.destroy(root.product.organization.label, root.content_id)') 
+            time.sleep(3)  
```

Comment:
```suggestion
    target_sat.execute(
        'echo "root = ::Katello::RootRepository.last; '
        '::Katello::Resources::Candlepin::Product.remove_content(root.product.organization.label, root.product.cp_id, root.content_id); '
        '::Katello::Resources::Candlepin::Content.destroy(root.product.organization.label, root.content_id)" | foreman-rake console'
    )
```
Instead of the `shell` and `sleep`s I would rather pipe the commands to the console this way.

Reply:
Or maybe this way, so that flake8 doen't complain about the long lines
```
    target_sat.execute(
        'echo "root = ::Katello::RootRepository.last; ::Katello::Resources::Candlepin::Product.'
        'remove_content(root.product.organization.label, root.product.cp_id, root.content_id); '
        '::Katello::Resources::Candlepin::Content.destroy(root.product.organization.label, '
        'root.content_id)" | foreman-rake console'
    )
```
Or use `# noqa` or whatever, it's up to you.